test/integration/setup.ts

```diff
@@ -36,61 +36,94 @@ export let noServiceAccountApp: admin.app.App;
 
 export let cmdArgs: any;
 
+export const isEmulator = !!process.env.FIREBASE_EMULATOR_HUB;
+export function skipForEmulator(): void {
+  before(function () {
+    if (isEmulator) {
+      this.skip();
+    }
+  });
+}
+
 before(() => {
-  /* tslint:disable:no-console */
-  let serviceAccount: any;
-  try {
-    serviceAccount = require('../resources/key.json');
-  } catch (error) {
-    console.log(chalk.red(
-      'The integration test suite requires a service account JSON file for a ' +
-      'Firebase project to be saved to `test/resources/key.json`.',
-      error,
-    ));
-    throw error;
-  }
+  let getCredential: () => {credential?: admin.credential.Credential};
+  let serviceAccountId: string;
 
-  try {
-    apiKey = fs.readFileSync(path.join(__dirname, '../resources/apikey.txt')).toString().trim();
-  } catch (error) {
-    console.log(chalk.red(
-      'The integration test suite requires an API key for a ' +
-      'Firebase project to be saved to `test/resources/apikey.txt`.',
-      error,
+  /* tslint:disable:no-console */
```

Comment:
Same here. We probably need a separate emulator-only test suite and the corresponding changes in CI to run it regularly.

Reply:
See comments above.

Reply:
https://github.com/firebase/firebase-admin-node/pull/1155